### ISSUE TITLE
The projectWizard fragment generates invalid code if the grammar does not use Xbase

### ISSUE THREAD

**Issue by LorenzoBettini** [eclipse/xtext-core#1318]
I was adding the new generator `projectFragment { generate = true }` to a new DSL, let's call it `MyDsl` (which does not inherit from Xbase) and when running MWE2 the created `MyDslProjectTemplateProvider.xtend` in the UI project does not compile (from what I understand, the active annotation `@ProjectTemplate` is not processed at all and the generated Java code for `HelloWorldProject` has no superclass); moreover, the additional `Messages.java` and `messages.properties` are not generated at all... when making the grammar extend Xbase, instead, running the MWE2 again fixes all the above issues.

**Comment by LorenzoBettini** [comment]
@miklossy can it be due to the additional dependencies in the `MANIFEST` of the UI project

```
 org.eclipse.xtext.common.types.ui,
 org.eclipse.xtext.xbase.ui,
 org.eclipse.jdt.debug.ui
```

which are generated only when switching to Xbase?

Note that after everything is generated correctly, removing the `MyDslProjectTemplateProvider.xtend` file, removing the inheritance from Xbase in the grammar, and running MWE2 again still works.

**Comment by LorenzoBettini** [comment]
So it looks like the required bundle in the `MANIFEST` to let the Xtend compile is `org.eclipse.xtext.xbase.ui`

**Comment by LorenzoBettini** [comment]
I confirm that, starting from scratch (without extending Xbase in the grammar) but adding `org.eclipse.xtext.xbase.ui` to the `MANIFEST` of the UI project generates the project wizard stuff correctly (and it compiles fine)

**Comment by cdietrich** [comment]
@ArneDeutsch @miklossy any idea what happens here?

**Comment by cdietrich** [comment]
@LorenzoBettini  can it be that the actual problem is a missing depenendcy to xtend.lib.macro?

**Comment by LorenzoBettini** [comment]
@cdietrich I was about to say that! I think that's the cause but I had to go out and I couldn't verify that. However, as I said, it is straightforward to reproduce the problem with a fresh new project.

**Comment by miklossy** [comment]
I can confirm that the problem exists and that can be solved by adding the `org.eclipse.xtend.lib` to the Require-Bundle section of the UI Plugin:
![screenshot](https://user-images.githubusercontent.com/20393472/70434078-8acdc300-1a83-11ea-8d9f-9ed6af13ccc8.png)

**Comment by cdietrich** [comment]
Fixed in 2.21

**Comment by LorenzoBettini** [comment]
@miklossy  actually xtend.lib.macro is enough to fix the problem

**Comment by miklossy** [comment]
I also encountered this problem together with @prggz while adding new project wizard / new file wizard to the GEF DOT editor ([561084](https://bugs.eclipse.org/bugs/show_bug.cgi?id=561084)). The fix for this problem is already provided by @cdietrich for the project wizard, but I think the same fix should be also applied to the file wizard as well in the [TemplateFileWizardFragment](https://github.com/eclipse/xtext-core/blob/12a21c9114d7600dae5029d38149e204d87f885b/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/fileWizard/TemplateFileWizardFragment.xtend#L61).

@cdietrich If you agree, could you come up with a corresponding PR?

**Comment by cdietrich** [comment]
@miklossy which fix do you mean?

**Comment by miklossy** [comment]
I mean this one https://github.com/eclipse/xtext-core/pull/1320

**Comment by cdietrich** [comment]
Second problem fixed in 2.22
